lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -124,29 +136,14 @@ sub run {
 
     # If the profile parameter has been set in the API, then load a profile
     if ( $params->{profile} ) {
-        if ( $params->{profile} eq 'test_profile_1' and Zonemaster::Backend::Config->CustomProfilesPath()) {
-            # The config has defined an alternative profile and "test_profile_1" has been set.
-            Zonemaster::Engine->config->load_policy_file( Zonemaster::Backend::Config->CustomProfilesPath() . '/policy.json' );
-        }
-        else { # The profile parameter has been set to something else or alternative profile is not defined
-            Zonemaster::Engine->config->load_policy_file( 'policy.json' );
-        }
-    # It will be silently ignored if the file does not exist.
-    }
-
-
-    if ( $params->{config} ) {
-        my $config_file_path = Zonemaster::Backend::Config->GetCustomConfigParameter('ZONEMASTER', $params->{config});
-        if ($config_file_path) {
-            if (-e $config_file_path) {
-                Zonemaster::Engine->config->load_config_file( $config_file_path );
-            }
-            else {
-                die "The file specified by the config parameter value: [$params->{config}] doesn't exist";
-            }
+        $params->{profile} = lc($params->{profile});
+        if (defined $self->{profiles}->{$params->{profile}} && $self->{profiles}->{$params->{profile}}->{zm_profile}) { 
+            my $profile = Zonemaster::Engine::Profile->default;
+            $profile->merge( $self->{profiles}->{$params->{profile}}->{zm_profile} );
+            Zonemaster::Engine::Profile->effective->merge( $profile );
         }
         else {
-            die "Unknown test configuration: [$params->{config}]\n"
+            die "Can't load profile: [$params->{profile}]" if ($params->{profile} ne 'default')
```

Comment:
This message is somewhat misleading since no profile loading is attempted at this point. Instead it should probably say something along the lines of "Unknown profile".